consensus/src/block_storage/block_store.rs

```diff
@@ -275,6 +275,12 @@ impl BlockStore {
 
         assert!(!blocks_to_commit.is_empty());
 
+        for block in &blocks_to_commit {
+            // Given the block is ready to commit, then it certainly must have a QC.
+            // However, we keep it an Option to be safe.
+            block.set_qc(self.get_quorum_cert_for_block(block.id()));
```

Comment:
instead of this, we can just do a reverse iteration since every block carries the parent's qc, we still need to get the qc for the last block though

Reply:
do you think the call is expensive? the change looks more verbose... I also want to Arc the QC so I can avoid the cloning the QC.


```
blocks_to_commit
            .last()
            .expect("at least one block is required")
            .set_qc(
                self
                    .get_quorum_cert_for_block(block.id())
                    .expect("QC must be present"),
            );

        for [parent, child] in blocks_to_commit.windows(2).rev() {
            parent.set_qc(Arc::new(child.block().quorum_cert().clone()));
        }
```

Reply:
nah I just didn't like the Option there

Reply:
Okay, i will just expect a QC then and still use the function call.